zerver/lib/zcommand.py

```diff
@@ -1,14 +1,27 @@
 from typing import Any, Dict
 from django.utils.translation import ugettext as _
 
-from zerver.models import UserProfile
+from zerver.models import UserProfile, get_realm_stream, \
+    get_stream_recipient, topic_exists, Stream
 from zerver.lib.actions import do_set_user_display_setting
 from zerver.lib.exceptions import JsonableError
+from zerver.lib.bugdown import possible_linked_stream_names
+
+def get_topic_from_mute_topic_content(content: str, stream_name: str) -> str:
+    """
+    This functions helps to extract the topic name from the mute_topic command
+    """
+    stream_link = ("#**%s**" % (stream_name))
+    topic_name = content.replace('mute_topic', '').replace(stream_link, '')
```

Comment:
This feels like a slightly brittle way to get the topic.  I feel we want a single function that uses a regex to get both values, where the regex looks something like `'#\*\*(.*)\*\* (.*)`.

Reply:
I am already using [possible_linked_stream_names](https://github.com/rheaparekh/zulip/blob/01ec7fd14bc6573b99330c964b606a1902855799/zerver/lib/zcommand.py#L49)  from bugdown to extract the stream name. What I am doing here is removing both, the stream mention and `mute_topic` from the command, hence we would have only topic left in the command.